src/boilerpipe/extract/__init__.py

```diff
@@ -42,6 +42,16 @@ def __init__(self, extractor='DefaultExtractor', **kwargs):
             if encoding.lower() == 'text/html':
                 encoding = chardet.detect(self.data)['encoding']
             try:
+                import gzip
```

Comment:
`gzip` should only be invoked when we are sure that the data is actually compressed. You may want to check an appropriate header.